cl/cl_chain.cc

```diff
@@ -218,7 +218,7 @@ void cl_chain_append(
 {
     try {
         ICodeListener *listener = cl_obtain_from_wrap(self);
-        ClChain *chain = dynamic_cast<ClChain *>(listener);
+        ClChain *chain = static_cast<ClChain *>(listener);
```

Comment:
Does not it indicate some problem with compiler flags?  Perhaps missing RTTI support?

Anyway, the error message for the subsequent  condition `(!chain)` needs to be updated (because `static_cast` never fails at run time).

Reply:
```
2019-04-20 17:02:27.895012+0200 cc1[41185:895562] dynamic_cast error 1: Both of the following type_info's should have public visibility.  At least one of them is hidden. 
```
RTTI works OK with gcc, in shared library doesn't. `-frtti` is enabled, I try set default visibility, then add option `-fvisibility-inlines-hidden`.
Honestly, I don't know. This applies to all `dynamic_cast`s, but predator works correctly (?!) And each throwing exception ends with abort (no catching unknown exception, all of them) and only, if it's part of shared library.